### ISSUE TITLE
warning from perl faq that strips C and C++ comments

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6656]
Migrated from [rt.perl.org#23131](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=23131) (status was 'resolved')

Searchable as RT23131$

**Comment by p5pRT** [comment]
### From david.dyck@fluke.com

#### Created by david.dyck@fluke.com



I would think that the examples in the perl faq
should be "warning clean"\, unless otherwize noted\,
or obvious from the context?

When I use the code directly from the perlfaq to remove comments
\(including C\+\+ comments\) from perlfaq6\.pod
 s\#/\\\*\[^\*\]\*\\\*\+\(\[^/\*\]\[^\*\]\*\\\*\+\)\*/|//\[^\\n\]\*|\("\(\\\\\.|\[^"\\\\\]\)\*"|'\(\\\\\.|\[^'\\\\\]\)\*'|\.\[^/"'\\\\\]\*\)\#$2\#gs;
as the source of the perl program using perl \-w \( I actually used perl
\-pwl and use the expression as the whole program\) I get warnings when
it removes both C and C\+\+ comments from input lines\.

My work around is to add a
&nbsp;    no warnings qw\(uninitialized\);
line around the substitution that triggers the warning\.  

Even the Fred Curtis's expanded expression triggers the warning\.

Perhaps a note in that faq to suggest that the user may
want to add the above no warnings line would be appropriate\.

<details><summary>Perl Info</summary>

```

Flags:
    category=docs
    severity=medium

Site configuration information for perl v5.9.0:

Configured by dcd at Tue Jul 22 08:31:10 PDT 2003.

Summary of my perl5 (revision 5.0 version 9 subversion 0 patch 20179) configuration:
  Platform:
    osname=linux, osvers=2.4.22-pre7, archname=i686-linux
    uname='linux dd 2.4.22-pre7 #4 sat jul 19 18:40:10 pdt 2003 i686 '
    config_args='-Dmksymlinks -Dinstallusrbinperl -Uversiononly -Dusedevel -Doptimize=-O3 -g -de -Dcf_email=david.dyck@fluke.com'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef useithreads=undef usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-DDEBUGGING -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O3 -g',
    cppflags='-DDEBUGGING -fno-strict-aliasing -I/usr/local/include'
    ccversion='', gccversion='egcs-2.91.66.1 19990314/Linux (egcs-1.1.2 release)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lgdbm -ldbm -ldb -ldl -lm -lc
    perllibs=-ldl -lm -lc
    libc=/lib/libc.so.5.4.44, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    DEVEL20173


@INC for perl v5.9.0:
    /usr/local/lib/perl5/5.9.0/i686-linux
    /usr/local/lib/perl5/5.9.0
    /usr/local/lib/perl5/site_perl/5.9.0/i686-linux
    /usr/local/lib/perl5/site_perl/5.9.0
    /usr/local/lib/perl5/site_perl/5.8.0/i686-linux
    /usr/local/lib/perl5/site_perl/5.8.0
    /usr/local/lib/perl5/site_perl
    .


Environment for perl v5.9.0:
    HOME=/home/dcd
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/dcd/bin:/sbin:/usr/local/bin:/bin:/usr/bin:/usr/X11/bin:/usr/games:/usr/local/samba:/home/hobbes/tools/scripts:/home/hobbes/tools/linux:/usr0/hobbes/tools/scripts:/usr0/dcd/bin:/apps/general/bin:/usr/public
    PERL5_CPANPLUS_CONFIG=/home/dcd/.cpanplus/config
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @eserte

\[dcd \- Sat Jul 26 11&#8203;:38&#8203;:47 2003\]&#8203;:

> 
> This is a bug report for perl from david\.dyck \<\!\-\-c\-\-> \<i>at\</i>
\<\!\-\-a\-\-> fluke\.com\,
> generated with the help of perlbug 1\.34 running under perl v5\.9\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> I would think that the examples in the perl faq
> should be "warning clean"\, unless otherwize noted\,
> or obvious from the context?
> 
> When I use the code directly from the perlfaq to remove comments
> \(including C\+\+ comments\) from perlfaq6\.pod
> 
s\#/\\\*\[^\*\]\*\\\*\+\(\[^/\*\]\[^\*\]\*\\\*\+\)\*/|//\[^\\n\]\*|\("\(\\\\\.|\[^"\\\\\]\)\*"|'\(\\\\\.|\[^'\\\\\]\)\*'|\.\[^/"'\\\\\]\*\)\#$2\#gs;
> as the source of the perl program using perl \-w \( I actually used perl
> \-pwl and use the expression as the whole program\) I get warnings when
> it removes both C and C\+\+ comments from input lines\.
> 
> My work around is to add a
>     no warnings qw\(uninitialized\);
> line around the substitution that triggers the warning\.
> 
> Even the Fred Curtis's expanded expression triggers the warning\.
> 
> Perhaps a note in that faq to suggest that the user may
> want to add the above no warnings line would be appropriate\.
> 

Or just check the definedness of $2\, like in the attached patch\.
This also fixes a missing ";" in the first code example\.

Regards\,
&nbsp;    Slaven



> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>     category=docs
>     severity=medium
> \-\-\-
> Site configuration information for perl v5\.9\.0&#8203;:
> 
> Configured by dcd at Tue Jul 22 08&#8203;:31&#8203;:10 PDT 2003\.
> 
> Summary of my perl5 \(revision 5\.0 version 9 subversion 0 patch 20179\)
>    configuration&#8203;:
>   Platform&#8203;:
>     osname=linux\, osvers=2\.4\.22\-pre7\, archname=i686\-linux
>     uname='linux dd 2\.4\.22\-pre7 \#4 sat jul 19 18&#8203;:40&#8203;:10 pdt 2003 i686 '
>     config\_args='\-Dmksymlinks \-Dinstallusrbinperl \-Uversiononly
>    \-Dusedevel \-Doptimize=\-O3 \-g \-de \-Dcf\_email=david\.dyck \<\!\-\-c\-\->
\<i>at\</i> \<\!\-\-a\-\-> fluke\.com'
>     hint=recommended\, useposix=true\, d\_sigaction=define
>     usethreads=undef useithreads=undef usemultiplicity=undef
>     useperlio=define d\_sfio=undef uselargefiles=define usesocks=undef
>     use64bitint=undef use64bitall=undef uselongdouble=undef
>     usemymalloc=n\, bincompat5005=undef
>   Compiler&#8203;:
>     cc='cc'\, ccflags ='\-DDEBUGGING \-fno\-strict\-aliasing
>    \-I/usr/local/include \-D\_LARGEFILE\_SOURCE \-D\_FILE\_OFFSET\_BITS=64'\,
>     optimize='\-O3 \-g'\,
>     cppflags='\-DDEBUGGING \-fno\-strict\-aliasing \-I/usr/local/include'
>     ccversion=''\, gccversion='egcs\-2\.91\.66\.1 19990314/Linux \(egcs\-
>    1\.1\.2 release\)'\, gccosandvers=''
>     intsize=4\, longsize=4\, ptrsize=4\, doublesize=8\, byteorder=1234
>     d\_longlong=define\, longlongsize=8\, d\_longdbl=define\,
>    longdblsize=12
>     ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\,
>    lseeksize=4
>     alignbytes=4\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='cc'\, ldflags =' \-L/usr/local/lib'
>     libpth=/usr/local/lib /lib /usr/lib
>     libs=\-lgdbm \-ldbm \-ldb \-ldl \-lm \-lc
>     perllibs=\-ldl \-lm \-lc
>     libc=/lib/libc\.so\.5\.4\.44\, so=so\, useshrplib=false\,
>    libperl=libperl\.a
>     gnulibc\_version=''
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-
>    rdynamic'
>     cccdlflags='\-fpic'\, lddlflags='\-shared \-L/usr/local/lib'
> 
> Locally applied patches&#8203;:
>     DEVEL20173
> 
> \-\-\-
> @&#8203;INC for perl v5\.9\.0&#8203;:
>     /usr/local/lib/perl5/5\.9\.0/i686\-linux
>     /usr/local/lib/perl5/5\.9\.0
>     /usr/local/lib/perl5/site\_perl/5\.9\.0/i686\-linux
>     /usr/local/lib/perl5/site\_perl/5\.9\.0
>     /usr/local/lib/perl5/site\_perl/5\.8\.0/i686\-linux
>     /usr/local/lib/perl5/site\_perl/5\.8\.0
>     /usr/local/lib/perl5/site\_perl
>     \.
> 
> \-\-\-
> Environment for perl v5\.9\.0&#8203;:
>     HOME=/home/dcd
>     LANG \(unset\)
>     LANGUAGE \(unset\)
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
>    
PATH=/home/dcd/bin&#8203;:/sbin&#8203;:/usr/local/bin&#8203;:/bin&#8203;:/usr/bin&#8203;:/usr/X11/bin&#8203;:/usr/games&#8203;:/usr/local/samba&#8203;:/home/hobbes/tools/scripts&#8203;:/home/hobbes/tools/linux&#8203;:/usr0/hobbes/tools/scripts&#8203;:/usr0/dcd/bin&#8203;:/apps/general/bin&#8203;:/usr/public
>     PERL5\_CPANPLUS\_CONFIG=/home/dcd/\.cpanplus/config
>     PERL\_BADLANG \(unset\)
>     SHELL=/bin/bash

**Comment by p5pRT** [comment]
### From @eserte

[perlfaq.pod.diff](https://rt-archive.perl.org/perl5/Ticket/Attachment/61920/45642/perlfaq.pod.diff)

**Comment by p5pRT** [comment]
### From peter.obermayer@siemens.com

Hello\, Slaven\,

I've got the same problem as the original submitter\.
On your e\-mail \(below\) you mention an attached patch\, which I do not have access to\.
Could I get a copy of it?

Best regards
Peter

Newsgroups&#8203;: perl\.perlfaq\.workers
Date&#8203;: 29 Jul 2003 13&#8203;:24&#8203;:02 \-0000
Subject&#8203;: \[perl \#23131\] warning from perl faq that strips C and C\+\+ comments 
Message\-ID&#8203;: \<rt\-23131\-61920\.16\.730211193614\[at\]rt\.perl\.org>
Cc&#8203;: perlfaq\-workers\[at\]perl\.org
To&#8203;: "OtherRecipients of perl Ticket \#23131"&#8203;: ;
From&#8203;: perlbug\-followup\[at\]perl\.org \(Slaven Rezic\)
References&#8203;: \<rt\-23131\[at\]perl> 


\[dcd \- Sat Jul 26 11&#8203;:38&#8203;:47 2003\]&#8203;:

> 
> This is a bug report for perl from david\.dyck \<\!\-\-c\-\-> \<i>at\</i>
\<\!\-\-a\-\-> fluke\.com\,
> generated with the help of perlbug 1\.34 running under perl v5\.9\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> I would think that the examples in the perl faq
> should be "warning clean"\, unless otherwize noted\,
> or obvious from the context?
> 
> When I use the code directly from the perlfaq to remove comments
> \(including C\+\+ comments\) from perlfaq6\.pod
> 
s\#/\\\*\[^\*\]\*\\\*\+\(\[^/\*\]\[^\*\]\*\\\*\+\)\*/|//\[^\\n\]\*|\("\(\\\\\.|\[^"\\\\\]\)\*"|'\(\\\\\.|\[^'\\\\\]\)\*'|\.\[^/"'\\\\\]\*\)\#$2\#gs;
> as the source of the perl program using perl \-w \( I actually used perl
> \-pwl and use the expression as the whole program\) I get warnings when
> it removes both C and C\+\+ comments from input lines\.
> 
> My work around is to add a
>     no warnings qw\(uninitialized\);
> line around the substitution that triggers the warning\.
> 
> Even the Fred Curtis's expanded expression triggers the warning\.
> 
> Perhaps a note in that faq to suggest that the user may
> want to add the above no warnings line would be appropriate\.
> 

Or just check the definedness of $2\, like in the attached patch\.
This also fixes a missing ";" in the first code example\.

Regards\,
&nbsp;    Slaven







Mit freundlichem Gruß
Dr\. Peter E\. Obermayer
Siemens VDO Automotive AG
Infotainment Solutions
Philipsstraße 1
D\-35576 Wetzlar
Tel\. \+49 6441 370 750
Fax \+49 6441 370 8326
E\-mail&#8203;: peter\.obermayer@&#8203;Siemens\.com

**Comment by p5pRT** [comment]
### From david.dyck@fluke.com


I've re\-attached the patch that
&nbsp;  Slaven Rezic \<perlbug\-followup@&#8203;perl\.org>
sent in the original that I received\.

Can you see it ok now?

On Mon\, 25 Oct 2004 at 16&#8203;:20 \-0000\, Obermayer Peter\-Ernst via RT\.\.\.&#8203;:

> From&#8203;: Obermayer Peter\-Ernst via RT \<perlbug\-followup@&#8203;perl\.org>
> To&#8203;: david\.dyck@&#8203;fluke\.com
> Date&#8203;: 25 Oct 2004 16&#8203;:20&#8203;:45 \-0000
> Subject&#8203;: \[perl \#23131\] warning from perl faq that strips C and C\+\+
>     comment s
>
> Hello\, Slaven\,
>
> I've got the same problem as the original submitter\.
> On your e\-mail \(below\) you mention an attached patch\, which I do not have access to\.
> Could I get a copy of it?
>
> Best regards
> Peter
>
> Newsgroups&#8203;: perl\.perlfaq\.workers
> Date&#8203;: 29 Jul 2003 13&#8203;:24&#8203;:02 \-0000
> Subject&#8203;: \[perl \#23131\] warning from perl faq that strips C and C\+\+ comments
> Message\-ID&#8203;: \<rt\-23131\-61920\.16\.730211193614\[at\]rt\.perl\.org>
> Cc&#8203;: perlfaq\-workers\[at\]perl\.org
> To&#8203;: "OtherRecipients of perl Ticket \#23131"&#8203;: ;
> From&#8203;: perlbug\-followup\[at\]perl\.org \(Slaven Rezic\)
> References&#8203;: \<rt\-23131\[at\]perl>
>
>
> \[dcd \- Sat Jul 26 11&#8203;:38&#8203;:47 2003\]&#8203;:
>
> >
> > This is a bug report for perl from david\.dyck \<\!\-\-c\-\-> \<i>at\</i>
> \<\!\-\-a\-\-> fluke\.com\,
> > generated with the help of perlbug 1\.34 running under perl v5\.9\.0\.
> >
> >
> > \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> > \[Please enter your report here\]
> >
> > I would think that the examples in the perl faq
> > should be "warning clean"\, unless otherwize noted\,
> > or obvious from the context?
> >
> > When I use the code directly from the perlfaq to remove comments
> > \(including C\+\+ comments\) from perlfaq6\.pod
> >
> s\#/\\\*\[^\*\]\*\\\*\+\(\[^/\*\]\[^\*\]\*\\\*\+\)\*/|//\[^\\n\]\*|\("\(\\\\\.|\[^"\\\\\]\)\*"|'\(\\\\\.|\[^'\\\\\]\)\*'|\.\[^/"'\\\\\]\*\)\#$2\#gs;
> > as the source of the perl program using perl \-w \( I actually used perl
> > \-pwl and use the expression as the whole program\) I get warnings when
> > it removes both C and C\+\+ comments from input lines\.
> >
> > My work around is to add a
> >     no warnings qw\(uninitialized\);
> > line around the substitution that triggers the warning\.
> >
> > Even the Fred Curtis's expanded expression triggers the warning\.
> >
> > Perhaps a note in that faq to suggest that the user may
> > want to add the above no warnings line would be appropriate\.
> >
>
> Or just check the definedness of $2\, like in the attached patch\.
> This also fixes a missing ";" in the first code example\.
>
> Regards\,
>     Slaven
>
> Mit freundlichem GruÃÂ
> Dr\. Peter E\. Obermayer
> Siemens VDO Automotive AG
> Infotainment Solutions
> PhilipsstraÃÂe 1
> D\-35576 Wetzlar
> Tel\. \+49 6441 370 750
> Fax \+49 6441 370 8326
> E\-mail&#8203;: peter\.obermayer@&#8203;Siemens\.com

**Comment by p5pRT** [comment]
### From david.dyck@fluke.com

<details><summary>perlfaq.pod.diff</summary>

```diff
--- perlfaq6.pod.orig	2003-07-29 15:16:48.000000000 +0200
+++ perlfaq6.pod	2003-07-29 15:19:52.000000000 +0200
@@ -342,7 +342,7 @@ created by Jeffrey Friedl and later modi
 
     $/ = undef;
     $_ = <>;
-    s#/\*[^*]*\*+([^/*][^*]*\*+)*/|("(\\.|[^"\\])*"|'(\\.|[^'\\])*'|.[^/"'\\]*)#$2#gs
+    s#/\*[^*]*\*+([^/*][^*]*\*+)*/|("(\\.|[^"\\])*"|'(\\.|[^'\\])*'|.[^/"'\\]*)#defined $2 ? $2 : ""#gse;
     print;
 
 This could, of course, be more legibly written with the C</x> modifier, adding
@@ -383,11 +383,11 @@ whitespace and comments.  Here it is exp
          .           ##  Anything other char
          [^/"'\\]*   ##  Chars which doesn't start a comment, string or escape
        )
-     }{$2}gxs;
+     }{defined $2 ? $2 : ""}gxse;
 
 A slight modification also removes C++ comments:
 
-    s#/\*[^*]*\*+([^/*][^*]*\*+)*/|//[^\n]*|("(\\.|[^"\\])*"|'(\\.|[^'\\])*'|.[^/"'\\]*)#$2#gs;
+    s#/\*[^*]*\*+([^/*][^*]*\*+)*/|//[^\n]*|("(\\.|[^"\\])*"|'(\\.|[^'\\])*'|.[^/"'\\]*)#defined $2 ? $2 : ""#gse;
 
 =head2 Can I use Perl regular expressions to match balanced text?
 

```
</details>

**Comment by p5pRT** [comment]
### From @tux

On Mon 25 Oct 2004 18&#8203;:28\, David Dyck \<david\.dyck@&#8203;fluke\.com> wrote&#8203;:
> I've re\-attached the patch that
>   Slaven Rezic \<perlbug\-followup@&#8203;perl\.org>
> sent in the original that I received\.
> 
> Can you see it ok now?

Ahhh\, that calls for defined\-or\!

s\#/\\\*\[^\*\]\*\\\*\+\(\[^/\*\]\[^\*\]\*\\\*\+\)\*/|//\[^\\n\]\*|\("\(\\\\\.|\[^"\\\\\]\)\*"|'\(\\\\\.|\[^'\\\\\]\)\*'|\.\[^/"'\\\\\]\*\)\#$2//""\#gse;

if only to increase the \(double\) obfuscation \(since // also is the start of
C\+\+ comments\, this might make readers think of it that way \<grin>\)

\-\- 
H\.Merijn Brand        Amsterdam Perl Mongers \(http&#8203;://amsterdam\.pm\.org/\)
using perl\-5\.6\.1\, 5\.8\.3\, & 5\.9\.x\, and 809 on  HP\-UX 10\.20 & 11\.00\, 11i\,
&nbsp;   AIX 4\.3\, SuSE 9\.0\, and Win2k\.           http&#8203;://www\.cmve\.net/~merijn/
http&#8203;://archives\.develooper\.com/daily\-build@&#8203;perl\.org/   perl\-qa@&#8203;perl\.org
send smoke reports to&#8203;: smokers\-reports@&#8203;perl\.org\, QA&#8203;: http&#8203;://qa\.perl\.org

**Comment by p5pRT** [comment]
### From comdog@panix.com

\[\[ This message was both posted and mailed&#8203;: see
&nbsp;   the "To\," "Cc\," and "Newsgroups" headers for details\. \]\]

In article \<Pine\.LNX\.4\.51\.0410250925470\.28669@&#8203;dd\.tc\.fluke\.com>\, David
Dyck \<david\.dyck@&#8203;fluke\.com> wrote&#8203;:

> I've re\-attached the patch that
>   Slaven Rezic \<perlbug\-followup@&#8203;perl\.org>
> sent in the original that I received\.

I applied the patch to the perlfaq repository\, but I had to do it
by hand since you were using an old version of the perlfaq\. :\)

\-\- 
brian d foy\, comdog@&#8203;panix\.com

**Comment by p5pRT** [comment]
### From dcd@fluke.com

On Mon\, 25 Oct 2004\, brian d foy via RT wrote&#8203;:

> \[\[ This message was both posted and mailed&#8203;: see
>   the "To\," "Cc\," and "Newsgroups" headers for details\. \]\]
>
> In article \<Pine\.LNX\.4\.51\.0410250925470\.28669@&#8203;dd\.tc\.fluke\.com>\, David
> Dyck \<david\.dyck@&#8203;fluke\.com> wrote&#8203;:
>
>> I've re\-attached the patch that
>>   Slaven Rezic \<perlbug\-followup@&#8203;perl\.org>
>> sent in the original that I received\.
>
> I applied the patch to the perlfaq repository\, but I had to do it
> by hand since you were using an old version of the perlfaq\. :\)


I didn't write the patch\, so I can't take credit\.

I reported  \[perl \#23131\]

Slaven Rezic sent a patch \(back in  29 Jul 2003\, but
I see hasn't even been applied to perl 5\.9\.2 yet\.\)

Obermayer Peter\-Ernst asked if I could have
a copy of the patch\, which I found in my email

H\.Merijn Brand \<h\.m\.brand@&#8203;hccnet\.nl>
noticed that the
&nbsp;  defined $2 ? $2 : ""
could now be stated as
&nbsp;  $2//""

I see the Defined\-or patch has itself
been around since 2002

&nbsp;    Change 17682 by hv@&#8203;hv\-crypt\.org on 2002/08/05 15&#8203;:04&#8203;:54
&nbsp;         Subject&#8203;: Defined\-or patch \(cleaned up\)

but is it fair game to use in the faq?

**Comment by p5pRT** [comment]
### From @smpeters

> \[comdog@&#8203;panix\.com \- Mon Oct 25 13&#8203;:42&#8203;:12 2004\]&#8203;:
> 
> \[\[ This message was both posted and mailed&#8203;: see
>    the "To\," "Cc\," and "Newsgroups" headers for details\. \]\]
> 
> In article \<Pine\.LNX\.4\.51\.0410250925470\.28669@&#8203;dd\.tc\.fluke\.com>\, David
> Dyck \<david\.dyck@&#8203;fluke\.com> wrote&#8203;:
> 
> > I've re\-attached the patch that
> >   Slaven Rezic \<perlbug\-followup@&#8203;perl\.org>
> > sent in the original that I received\.
> 
> I applied the patch to the perlfaq repository\, but I had to do it
> by hand since you were using an old version of the perlfaq\. :\)
> 

These changes were included in bleadperl with change \#23466\.

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
